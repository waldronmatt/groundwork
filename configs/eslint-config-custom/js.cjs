module.exports = {
  env: {
    browser: true,
    node: true,
    es2020: true,
  },
  extends: ['eslint:recommended', 'plugin:import/recommended', 'airbnb-base'],
  rules: {
    'import/no-extraneous-dependencies': [
      'error',
      {
        devDependencies: true,
      },
    ],
  },
};
