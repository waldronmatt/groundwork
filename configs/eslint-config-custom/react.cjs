module.exports = {
  root: true,
  env: {
    browser: true,
    node: true,
    es2020: true,
  },
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint', 'react-refresh'],
  extends: [
    'eslint:recommended',
    'plugin:import/recommended',
    'plugin:@typescript-eslint/recommended-type-checked',
    'airbnb-typescript/base',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
  ],
  rules: {
    'import/extensions': 'off',
    'import/no-extraneous-dependencies': [
      'error',
      {
        devDependencies: true,
      },
    ],
    'react-refresh/only-export-components': ['warn', { allowConstantExport: true }],
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
};
