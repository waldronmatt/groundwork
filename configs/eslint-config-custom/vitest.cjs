module.exports = {
  globals: {
    vi: true,
  },
  // plugins: ['vitest'],
  // extends: ['plugin:vitest/recommended'],
};
