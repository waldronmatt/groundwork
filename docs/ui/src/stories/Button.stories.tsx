import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Button, type ButtonProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta = {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
  parameters: {
    badges: ['stable'],
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LKQ4FJ4bTnCSjedbRpk931/Sample-File',
    },
  },
} satisfies Meta<typeof Button>;

export default meta;

const getCaptionForLocale = (locale: string) => {
  switch (locale) {
    case 'es':
      return 'Hola!';
    default:
      return 'Hello!';
  }
};

const onClickHandler = action('button-click');

const Template: StoryFn<typeof Button> = (args: ButtonProps, { globals: { locale } }) => {
  // by default, let the user modify the text through the `children` control
  // otherwise, load in the text from locale for i18n testing
  const caption = locale !== '' ? getCaptionForLocale(locale as string) : args.children;
  return (
    <Button {...args} onClick={onClickHandler}>
      {caption}
    </Button>
  );
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});
export const Hover = Template.bind({});

Default.args = {
  ...defaultProps,
  variant: 'md',
};

Small.args = {
  ...defaultProps,
  variant: 'sm',
};

Large.args = {
  ...defaultProps,
  variant: 'lg',
};

Hover.args = {
  ...defaultProps,
  variant: 'md',
};

Hover.parameters = {
  pseudo: { hover: true },
};
