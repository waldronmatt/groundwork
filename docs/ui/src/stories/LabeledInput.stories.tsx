import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { LabeledInput, type LabeledInputProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta: Meta<typeof LabeledInput> = {
  title: 'Components/LabeledInput',
  component: LabeledInput,
  argTypes: {
    variant: {
      table: {
        disable: true,
      },
    },
  },
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LKQ4FJ4bTnCSjedbRpk931/Sample-File',
    },
  },
};

export default meta;

const onChangeHandler = action('labeledInput-change');
const onFocusHandler = action('labeledInput-focus');

const Template: StoryFn<typeof LabeledInput> = (args: LabeledInputProps) => {
  return <LabeledInput {...args} onChange={onChangeHandler} onFocus={onFocusHandler} />;
};

export const Default = Template.bind({});

Default.args = {
  ...defaultProps,
  id: 'unique_id',
};
