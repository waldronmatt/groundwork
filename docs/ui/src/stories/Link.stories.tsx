import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Link, type LinkProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
  href: '/',
};

const meta = {
  title: 'Components/Link',
  component: Link,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
  parameters: {
    badges: ['stable'],
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LKQ4FJ4bTnCSjedbRpk931/Sample-File',
    },
  },
} satisfies Meta<typeof Link>;

export default meta;

const getCaptionForLocale = (locale: string) => {
  switch (locale) {
    case 'es':
      return 'Hola!';
    default:
      return 'Hello!';
  }
};

const onClickHandler = action('link-click');

const Template: StoryFn<typeof Link> = (args: LinkProps, { globals: { locale } }) => {
  // by default, let the user modify the text through the `children` control
  // otherwise, load in the text from locale for i18n testing
  const caption = locale !== '' ? getCaptionForLocale(locale as string) : args.children;
  return (
    <Link
      {...args}
      onClick={(event) => {
        event.preventDefault();
        onClickHandler(event);
      }}
    >
      {caption}
    </Link>
  );
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});

Default.args = {
  ...defaultProps,
  variant: 'md',
};

Small.args = {
  ...defaultProps,
  variant: 'sm',
};

Large.args = {
  ...defaultProps,
  variant: 'lg',
};
