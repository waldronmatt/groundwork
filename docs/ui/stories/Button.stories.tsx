import type { Meta, StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { Button, type ButtonProps } from '@waldronmatt/demo-ui/lib/index.js';

const defaultProps = {
  children: 'Hello World',
};

const meta: Meta<typeof Button> = {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
};

export default meta;

const onClickHandler = action('button-click');

const Template: StoryFn<typeof Button> = (args: ButtonProps) => {
  return <Button {...args} onClick={onClickHandler} />;
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});

Default.args = {
  ...defaultProps,
  variant: 'md',
};

Small.args = {
  ...defaultProps,
  variant: 'sm',
};

Large.args = {
  ...defaultProps,
  variant: 'lg',
};
