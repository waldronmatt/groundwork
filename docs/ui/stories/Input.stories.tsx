import type { Meta, StoryFn } from '@storybook/react';
// we use `lib/` paths so we can get source files and have storybook
// auto refresh (hmr) whenever we update our component source files
import { Input, type InputProps } from '@waldronmatt/demo-ui/lib/index.js';

const meta: Meta<typeof Input> = {
  title: 'Components/Input',
  component: Input,
  argTypes: {
    variant: {
      options: ['md', 'sm', 'lg'],
      control: { type: 'radio' },
    },
  },
};

export default meta;

const Template: StoryFn<typeof Input> = (args: InputProps) => {
  return <Input {...args}></Input>;
};

export const Default = Template.bind({});
export const Small = Template.bind({});
export const Large = Template.bind({});

Default.args = {
  variant: 'md',
};

Small.args = {
  variant: 'sm',
};

Large.args = {
  variant: 'lg',
};
