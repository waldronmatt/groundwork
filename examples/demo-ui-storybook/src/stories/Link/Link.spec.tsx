import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { Link } from '@waldronmatt/demo-ui/lib/index.js';
import { describe, it, expect, vi } from 'vitest';
import { composeStories } from '@storybook/react';
import { DefaultProps } from './Link.stories.js';
import * as stories from './Link.stories.js';

const { Default, Small, Large } = composeStories(stories);

describe('Link component', () => {
  it('renders with default props', () => {
    const { container } = render(<Default />);
    const linkElement = screen.getByText(DefaultProps.children);

    expect(linkElement).toBeInTheDocument();
    expect(linkElement.tagName).toBe('A');
    expect(linkElement).toHaveClass('link');
    expect(linkElement).toHaveClass('md');
    expect(linkElement).toHaveAttribute('href', DefaultProps.href);
    expect(container).toMatchSnapshot();
  });

  it('renders with sm variant', () => {
    const { container } = render(<Small />);
    const linkElement = screen.getByText(DefaultProps.children);

    expect(linkElement).toHaveClass('sm');
    expect(container).toMatchSnapshot();
  });

  it('renders with lg variant', () => {
    const { container } = render(<Large />);
    const linkElement = screen.getByText(DefaultProps.children);

    expect(linkElement).toHaveClass('lg');
    expect(container).toMatchSnapshot();
  });

  it('handles native props', () => {
    const { container } = render(<Link {...DefaultProps} target="_blank" />);
    const linkElement = screen.getByText(DefaultProps.children);

    expect(linkElement).toHaveAttribute('target', '_blank');
    expect(container).toMatchSnapshot();
  });

  it('handles click events', () => {
    const onClickMock = vi.fn();
    render(<Link {...DefaultProps} onClick={onClickMock} />);
    const linkElement = screen.getByText(DefaultProps.children);
    fireEvent.click(linkElement);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
