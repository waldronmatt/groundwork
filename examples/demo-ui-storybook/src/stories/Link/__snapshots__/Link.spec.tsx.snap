// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link component > handles native props 1`] = `
<div>
  <a
    class="link md"
    href="/"
    target="_blank"
  >
    Hello World
  </a>
</div>
`;

exports[`Link component > renders with default props 1`] = `
<div>
  <a
    class="link md"
    href="/"
  >
    Hello World
  </a>
</div>
`;

exports[`Link component > renders with lg variant 1`] = `
<div>
  <a
    class="link lg"
    href="/"
  >
    Hello World
  </a>
</div>
`;

exports[`Link component > renders with sm variant 1`] = `
<div>
  <a
    class="link sm"
    href="/"
  >
    Hello World
  </a>
</div>
`;
